File: app/src/main/java/com/ozhan/mustafa/howl/core/login/LoginContract.java
package com.ozhan.mustafa.howl.core.login;

import android.app.Activity;

/**
 * Created by Mustafa Özhan on 3/4/17 at 12:55 PM.
 */

public interface LoginContract {
    interface View {
        void onLoginSuccess(String message);

        void onLoginFailure(String message);
    }

    interface Presenter {
        void login(Activity activity, String email, String password);
    }

    interface Interactor {
        void performFirebaseLogin(Activity activity, String email, String password);
    }

    interface OnLoginListener {
        void onSuccess(String message);

        void onFailure(String message);
    }
}

File: app/src/main/java/com/ozhan/mustafa/howl/core/login/LoginContract.kt
package com.ozhan.mustafa.howl.core.login

import android.app.Activity

/**
 * Created by Mustafa Özhan on 3/4/17 at 12:55 PM.
 */

interface LoginContract {
    interface View {
        fun onLoginSuccess(message: String)

        fun onLoginFailure(message: String)
    }

    interface Presenter {
        fun login(activity: Activity, email: String, password: String)
    }

    interface Interactor {
        fun performFirebaseLogin(activity: Activity, email: String, password: String)
    }

    interface OnLoginListener {
        fun onSuccess(message: String)

        fun onFailure(message: String)
    }
}
